### PR TITLE
Add kafka backend

### PR DESCRIPTION
**Pull request by saryani**
This PR will make it possible for osctrl to send the osquery-agent logs to a kafka topic. this means that these logs will no longer be pushed to the database, you need to set kafka as the log backend at init time otherwise it will log into database. 

* added a logger backend to send the agents logs to kafka
* added required configuration definition to allow setting up a kafka logger
* updated the docker-compose to allow to run different backends

### CODE REVIEW

File: backend/backend.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 
 const (
 	// DBString to format connection string to database for postgres
-	DBString = "host=%s port=%s dbname=%s user=%s password=%s sslmode=disable"
+	DBString = "host=%s port=%s dbname=%s user=%s password=%s sslmode=require"
```
Comment (saryani) [inline backend/backend.go:15]: i will revert this line before creating PR against upstream, it's needed for our tests

Comment (saryani) [reply backend/backend.go:15]: I'm making some changes to make this configurable!

Comment (zhuoyuan-liu) [reply backend/backend.go:15]: We don't need this db change as the author already added them

Comment (saryani) [reply backend/backend.go:15]: oh maaaaan!

File: .env.example [hunk 2]
```diff
@@ -11,3 +11,18 @@ SESSION_KEY=sessionkey
 OSCTRL_USER=admin
 OSCTRL_PASS=Changeme123!
 GOLANG_VERSION=1.23.0-bookworm
+
+OSCTRL_TLS_LOGGER=db
+
+LOG_KAFKA_BOOTSTRAP_SERVERS=my-kafka-bootstrap-server
```
Comment (zhuoyuan-liu) [inline .env.example:17]: Need be disabled by default for these config and add more comments for it

File: docker-compose-dev.yml [hunk 3]
```diff
@@ -54,6 +54,15 @@ services:
       - DB_PASS=${POSTGRES_DB_PASSWORD}
       #### Redis settings ####
       - REDIS_HOST=osctrl-redis
+      #### Kafka settings ####
+      - LOG_KAFKA_BOOTSTRAP_SERVERS=${LOG_KAFKA_BOOTSTRAP_SERVERS}
```
Comment (zhuoyuan-liu) [inline docker-compose-dev.yml:58]: Same here, it should be disabled by default.

File: logging/kafka.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	ctx := context.Background()
+	key := []byte(uuid) //uuid is the unique id of the os-query agent host that sent this data
+	rec := kgo.Record{Topic: l.config.Topic, Key: key, Value: data}
+	l.producer.Produce(ctx, &rec, func(r *kgo.Record, err error) {
+		if err != nil {
+			log.Printf(
+				"failed to produce message to kafka topic '%s'. details: %s", l.config.Topic, err)
+		}
+		if debug {
+			log.Printf("message with key '%s' was sent to topic '%s' successfully\n%s", key, l.config.Topic, string(data))
```
Comment (zhuoyuan-liu) [inline logging/kafka.go:105]: I think they updated the log for the application, try to merge with the upstream

**Pull request merged**